### PR TITLE
LocalNav component

### PR DESCRIPTION
**Pull request by tyom**
- Add LocalNav macro
- Replace existing hard-coded local navs with macro on investment projects and contacts page

Depends on #343

### CODE REVIEW

File: src/apps/companies/views/_layout-view.njk [hunk 1]
```diff
@@ -21,15 +21,15 @@
 
 
 {% block main_grid_left_column %}
-  <nav class="local-nav">
-    <a class="local-nav__link {% if tab == 'details' %}is-active{% endif %}" href="{{companyUrl}}">Company details</a>
+  <nav class="c-local-nav">
```
Comment (teneightfive) [inline src/apps/companies/views/_layout-view.njk]: Don't you want to use the macro for this?

File: src/middleware/local-nav.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const path = require('path')
+
+function localNavMiddleware (items = []) {
+  return function buildLocalNav (req, res, next) {
+    res.locals.localNavItems = items.map(item => {
+      const url = path.resolve(res.locals.CURRENT_PATH, item.path)
+      return Object.assign(item, {
+        url,
+        isActive: res.locals.CURRENT_PATH === url,
```
Comment (teneightfive) [inline src/middleware/local-nav.js]: Should this not be done in the macro as it's only used for rendering purposes?

When setting a list of items to view locals it probably doesn't care about this.

File: src/apps/companies/views/_layout-view.njk (hunk 1 above)
Comment (tyom) [inline src/apps/companies/views/_layout-view.njk]: As I explained in hangouts it's not straightforward to use this macro for local nav in companies. It would have to be a separate PR.

File: src/middleware/local-nav.js (hunk 2 above)
Comment (tyom) [inline src/middleware/local-nav.js]: But this is a generic middleware which creates new `url` property from `path` and sets `isActive`. Otherwise macro would have to expose `path.resolve` to do this operation.

Comment (teneightfive) [inline src/middleware/local-nav.js]: Yeah that bit makes sense, but was wondering whether you need to set the `isActive` property within the middleware or whether that can be set within the macro as that's the only place which is concerned with it.

File: src/apps/companies/views/_layout-view.njk (hunk 1 above)
Comment (ztolley) [inline src/apps/companies/views/_layout-view.njk]: We so need to refactor company. I'll dig out my plan and try Again to convince that we need to do it and figure out how I can do it a piece at a time

File: src/middleware/local-nav.js (hunk 2 above)
Comment (tyom) [inline src/middleware/local-nav.js]: Fair enough. Updated.

Comment (tyom) [inline src/middleware/local-nav.js]: I changed my mind. I think it's better to avoid side effects in macros (relying on external properties) for the sake of its testability and encapsulation.
